src/model/SubSurface_Impl.hpp

```diff
@@ -186,6 +192,14 @@ namespace detail {
 
     void resetShadingControl();
 
+    bool setShadingControls(const std::vector<ShadingControl> &shadingControls);
+
+    bool insertShadingControl(unsigned shadingControlIndex, const ShadingControl &shadingControl);
```

Comment:
Same comment about order as above. I am not sure it's appropriate or even wanted?

Reply:
Is the idea to tie into the field "Shading Control Sequence Number" on the ShadingControl object? That field is a bit weird... 

https://github.com/NREL/OpenStudio/blob/fa9a77bf173556a2a044f64789a7f4f01b0df076/src/energyplus/ForwardTranslator/ForwardTranslateShadingControl.cpp#L114-L119